controllers/nodehealthcheck_controller_test.go

```diff
@@ -446,6 +446,55 @@ var _ = Describe("Node Health Check CR", func() {
 					Expect(underTest.Status.UnhealthyNodes).To(BeEmpty())
 				})
 			})
+
+			When("two NHC CRs with different templates target the same unhealthy node", func() {
+
+				otherTestCRName := "other-test"
+				var otherTestCR *v1alpha1.NodeHealthCheck
+
+				BeforeEach(func() {
+					// prepare other CR but do not create yet, in order to have a predictable order of things to happen
+					otherTestCR = underTest.DeepCopy()
+					otherTestCR.SetName(otherTestCRName)
+					otherTestCR.Spec.RemediationTemplate = &v1.ObjectReference{
+						Kind:       "Metal3RemediationTemplate",
+						Namespace:  MachineNamespace,
+						Name:       "ok",
+						APIVersion: "test.medik8s.io/v1alpha1",
+					}
+
+					setupObjects(1, 2, true)
+				})
+
+				It("only one NHC should remediate that node", func() {
+
+					By("Verifying node is remediated by 1st NHC")
+					Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+					Expect(underTest.Status.InFlightRemediations).To(HaveLen(1))
+					Expect(underTest.Status.UnhealthyNodes).To(HaveLen(1))
+					Expect(underTest.Status.UnhealthyNodes[0].Remediations).To(HaveLen(1))
+
+					By("Creating a 2nd NHC")
+					Expect(k8sClient.Create(context.Background(), otherTestCR)).To(Succeed())
```

Comment:
Nit: I think a more realistic test would be to create both NHCs at the same time and verify only one creates remediation (over the alternative of creating the second one after remediation is already created)

Reply:
What would be the difference? 🤔 I avoided it to make the test more simple because I know which one is going to remediate...

Reply:
This is a good point, the way I see it it's a tradeoff:
One the one hand we'd like tests to simulate the actual situation as closely as possible and on the other hand we want to keep tests simple and maintainable - those two values often clash.

Reply:
Ah, now I get what you're aiming at: check if we run into race conditions when things happen simultaneously.
That's a fair point... but, at least in this specific case: even if we run into a race condition and 2 CRs are created for the same node... there is probably nothing we can do about it, or it would at least make things way more complicated. So with the current straigtforward implementation, a better test would potentially just make the test flaky. So I keep it as it is. But good feedback, thanks!